core/elf.cc

```diff
@@ -1181,6 +1169,9 @@ std::string object::pathname()
 // Run the object's static constructors or similar initialization
 void object::run_init_funcs(int argc, char** argv)
 {
+    if (is_statically_linked()) {
+        return;
+    }
```

Comment:
Nitpick: you already if'ed around the call to run_init_funcs(), so no need to add an if() also inside the function?

Reply:
After looking at this again I do not think I "if" around the call to `run_init_funcs`, do I?